usecases/objects/references_add.go

```diff
@@ -69,8 +68,12 @@ func (m *Manager) AddObjectReference(ctx context.Context, principal *models.Prin
 	}
 
 	if !deprecatedEndpoint {
-		if input.Class != "" && strings.Count(string(input.Ref.Beacon), "/") == 3 {
-			toClass, toBeacon, replace, err := m.autodetectToClass(ctx, principal, input.Class, input.Property, input.Ref.Beacon)
+		beacon, err := crossref.Parse(input.Ref.Beacon.String())
+		if err != nil {
+			return &Error{"cannot parse beacon", StatusBadRequest, err}
+		}
```

Comment:
if beacon is used only when `input.Class != ""`, parsing can be moved to `if` block

Reply:
then I would need 2 different if blocks. Kept it like this for the sake of readability